bot-components/GitHub_queries.ml

```diff
@@ -267,6 +267,81 @@ let get_pull_request_refs ~bot_info ~owner ~repo ~number =
           f "Query pull_request_info failed with %s" err)
   >|= Result.bind ~f:(pull_request_info_of_resp ~owner ~repo ~number)
 
+let pull_request_reviews_info_of_resp ~owner ~repo ~number resp :
+    (GitHub_subscriptions.pull_request_reviews_info, string) Result.t =
+  match resp#repository with
+  | None ->
+      Error (f "Unknown repository %s/%s." owner repo)
+  | Some repository -> (
+    match repository#pullRequest with
+    | None ->
+        Error (f "Unknown pull request %s/%s#%d." owner repo number)
+    | Some pull_request -> (
+      match
+        ( pull_request#baseRef
+        , pull_request#files
+        , pull_request#approvedReviews
+        , pull_request#commentReviews
+        , pull_request#reviewDecision )
+      with
+      | None, _, _, _, _ ->
+          Error "No base ref found."
+      | _, None, _, _, _ ->
+          Error "No files found."
+      | _, _, None, _, _ ->
+          Error "No approved reviews found."
+      | _, _, _, None, _ ->
+          Error "No comment reviews found."
+      | _, _, _, _, None ->
+          Error "No review decision found."
+      | ( Some baseRef
+        , Some files
+        , Some approved_reviews
+        , Some comment_reviews
+        , Some review_decision ) ->
+          Ok
+            { baseRef= baseRef#name
+            ; files=
+                ( match files#nodes with
+                | None ->
+                    []
+                | Some files ->
+                    files |> Array.to_list |> List.filter_opt
+                    |> List.map ~f:(fun file -> file#path) )
+            ; approved_reviews=
+                ( match approved_reviews#nodes with
+                | None ->
+                    []
+                | Some reviews ->
+                    reviews |> Array.to_list |> List.filter_opt
+                    |> List.filter_map ~f:(fun review ->
+                           review#author
+                           |> Option.map ~f:(fun (`Actor a) -> a#login)) )
```

Comment:
We should add a deduplication step here:
```suggestion
                           |> Option.map ~f:(fun (`Actor a) -> a#login))
                           |> List.dedup_and_sort ~compare:String.compare )
```

Otherwise, we might end up listing reviewers several times.